### PR TITLE
Renaming of functions CubicInterpolation_DP and CubicInterpolation_MFLOW

### LINKED ISSUE #2006: Names of CubicInterpolation_DP() und CubicInterpolation_MFLOW() are misleading
**Reported by wischhusen on 6 Jun 2016 07:42 UTC**
Modelica.Fluid.Dissipation.Utilities.Functions.General.CubicInterpolation_DP() calculates a Reynolds number from other Re-inputs but does not take a pressure difference as input
Modelica.Fluid.Dissipation.Utilities.Functions.General.CubicInterpolation_MFLOW() calculates a lambda but does not take a mass flow as input

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/2006

### PR DESCRIPTION
**Pull request by wischhusen**
close #2006

### CODE REVIEW

File: Modelica/Fluid/Dissipation.mo [hunk 1]
```diff
@@ -10804,8 +10807,11 @@ The heterogeneous approaches are analytically derived by minimising the momentum
 
         algorithm
           lambda2 := 64*Re1*(Re/Re1)^(1 + dx*(c2 + dx*c3));
-          annotation (Inline=false, smoothOrder=5);
-        end CubicInterpolation_MFLOW;
+          annotation (Inline=false, smoothOrder=5,
+		  Documentation(revisions="<html>
+2018-11-20 Stefan Wischhusen: Renamed function from CubicInterpolation_MFLOW to CubicInterpolation_lambda.
+</html>");
```
Comment (beutlich) [inline Modelica/Fluid/Dissipation.mo]:
```suggestion
</html>"));
```

**Comment by beutlich** [comment]
Note, this only can go be merged to master branch (targeting the next major release), once MSL v3.2.3 is released.

**Comment by beutlich** [comment]
@wischhusen Can you please provide a unit test (say `ModelicaTest.Fluid.Dissipation.ConversionTest.'#2780'`) where conversion actually takes place? Thanks!

**Comment by wischhusen** [comment]
Added the conversion scirpt to Modelica/Resources/Scripts/Dymola and a Tester (Issue2780) to ModelicaTest.Fluid.Dissipation.ConversionTest. 

I'm a bit puzzled about the correct version numbers for the change. It should be 3.2.4, right?

**Comment by wischhusen** [comment]
Done

**Comment by dietmarw** [comment]
Please do not push branches to the MSL repository but use your own fork for that. Also the last correction is not present in the PR here. So you might have pushed to the wrong place.

**Comment by wischhusen** [comment]
There should be a branch issue2780. Anyhow, if that cannot be merged I will proceed with the fork.

**Comment by dietmarw** [comment]
Yes best to make a new PR based on a branch from your fork and also make sure that there is a test model in ModelicaTest that would trigger that conversion.

**Comment by beutlich** [comment]
> I was wondering if the location Scripts/Dymola is still the appropriate one since the conversion script is standardised for MLS 3.4 so in MSL 4.0.0 the standard script should probably be placed directly under Scripts?

See #2819 for the corresponding PR.

**Comment by beutlich** [comment]
Sorry for using this PR as guinea pig for the conversion. We will care for the proper location of the conversion script and the conversion tester.

File: Modelica/package.mo [hunk 2]
```diff
@@ -8360,12 +8360,7 @@ dateModified = "2019-01-23 12:00:00Z",
 revisionId="$Format:%h %ci$",
 uses(Complex(version="3.2.3"), ModelicaServices(version="3.2.3")),
 conversion(
- from(version="3.2.3", script="modelica://Modelica/Resources/Scripts/Conversion/ConvertModelica_from_3.2.3_to_4.0.0.mos"),
- noneFromVersion="3.2.1",
- noneFromVersion="3.2",
- noneFromVersion="3.1",
- noneFromVersion="3.0.1",
- noneFromVersion="3.0",
+ from(version={"3.0", "3.0.1", "3.1", "3.2", "3.2.1", "3.2.2", "3.2.3"}, script="modelica://Modelica/Resources/Scripts/Conversion/ConvertModelica_from_3.2.3_to_4.0.0.mos"),
```
Comment (dietmarw) [inline Modelica/package.mo]: Maybe place last in the sequence. Pet peeve.

**Comment by beutlich** [comment]
This is an important PR since it adds ConvertModelica_from_3.2.3_to_4.0.0.mos and ModelicaTestConversion4.mo, which all other isssues requiring conversion depend on.

@hubertus65 Can you please run the merge.

**Comment by beutlich** [comment]
> @hubertus65 Can you please run the merge.

Friendly reminder.

**Comment by hubertus65** [comment]
I have been traveling quite a bit. I'll try to find some block time for this. Any other tasks for me I missed?

**Comment by beutlich** [comment]
> Any other tasks for me I missed?

Yes, also #2816 and #2818 await your review.

**Comment by dietmarw** [comment]
> The minor issue is that conversion from MSL 2 could specify to="3.2.3"; so that the libraries using MSL 2 could be converted to MSL 4. (Which is super-rare and those conversions weren't changed anyway.)
I discussed this with @beutlich and decided against including MSL 2 conversions since as far as I know, most of the current tools don't even support MSL 2 anymore. So no point.

**Pull request merged**